app/uk/gov/hmrc/agentsubscriptionfrontend/models/AddressLookupAddress.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.models
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import cats.kernel.Monoid
+import play.api.data.validation.ValidationError
+import play.api.Play.current
+import play.api.i18n.Messages
+import play.api.i18n.Messages.Implicits._
+import play.api.libs.json.{OFormat, _}
+import uk.gov.hmrc.agentsubscriptionfrontend.config.blacklistedpostcodes.PostcodesLoader
+import uk.gov.hmrc.agentsubscriptionfrontend.models.AddressLookupAddress.ValidatedDesAddress
+import cats.implicits._
+
+import ValidatedDesAddress._
+
+case class Country(code: String,
+                   name: Option[String])
+
+case class AddressLookupAddress(lines: Seq[String],
+                                postcode: Option[String],
+                                country: Country)
+
+case class DesAddress(addressLine1: String,
+                      addressLine2: Option[String] = None,
+                      addressLine3: Option[String] = None,
+                      addressLine4: Option[String],
+                      postcode: Option[String],
+                      countryCode: String)
+
+object AddressLookupAddress {
+
+  implicit val formatCountry = Json.format[Country]
+  implicit val formatAddressLookupAddress = Json.format[AddressLookupAddress]
+
+  type ValidatedDesAddress[T] = Validated[Set[ValidationError], T]
+
+  private val postCodeRegex = "^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}$|BFPO\\s?[0-9]{1,5}$".r
+  private val desTextRegex = "^[A-Za-z0-9 \\-,.&'\\/]*$"
+  private val maxLength = 35
+
+  object ValidatedDesAddress {
+    implicit val validationResultMonoid = new Monoid[ValidatedDesAddress[DesAddress]] {
+      def empty: ValidatedDesAddress[DesAddress] = Valid(DesAddress("",None,None,None,None,""))
+
+      def combine(x: ValidatedDesAddress[DesAddress], y: ValidatedDesAddress[DesAddress]): ValidatedDesAddress[DesAddress] = (x, y) match {
+        case (Invalid(a), Invalid(b)) => Invalid(a ++ b)
+        case (Valid(a), Valid(b)) => Valid(b)
```

Comment:
This breaks the (Monoid laws)[https://en.wikibooks.org/wiki/Haskell/Monoids#Monoid_laws], specifically right identity. i.e. `validationResultMonoid` is not really a monoid.

Use of Monoid here also strikes me as overcomplicated and difficult to understand, can it be avoided?